### PR TITLE
make things immutable and nullable where possible

### PR DESCRIPTION
**Pull request by zeitlinger**
... to improve readability

remove @Autowired in constructors, which are not needed since boot 2

### CODE REVIEW

File: github-integration/src/main/java/de/zalando/zally/integration/zally/Violation.kt [hunk 1]
```diff
@@ -1,11 +1,10 @@
...
-        var title: String? = null,
-        var description: String? = null,
-        var violationType: ViolationType? = null,
-        var ruleLink: String? = null,
-        var paths: List<String>? = null
+    val title: String? = null,
+    val description: String? = null,
+    val violationType: ViolationType? = null,
+    val ruleLink: String? = null,
+    val paths: List<String>? = null
```
Comment (maxim-tschumak) [inline github-integration/src/main/java/de/zalando/zally/integration/zally/Violation.kt]: Let's initialize this also as an empty list.

**Comment by maxim-tschumak** [comment]
:+1:

**Comment by zeitlinger** [comment]
:+1:

**Comment by tkrop** [comment]
:+1:

**Comment by zeitlinger** [comment]
:+1:

**Comment by maxim-tschumak** [comment]
:+1:

**Pull request merged**